yucca/preprocessing/UnsupervisedPreprocessor.py

```diff
@@ -0,0 +1,138 @@
+"""
+Takes raw data conforming with Yucca standards and preprocesses according to the generic scheme
+"""
+import numpy as np
+import re
+import torch
+import nibabel as nib
+import os
+from yucca.preprocessing.YuccaPreprocessor import YuccaPreprocessor
+from yucca.paths import yucca_preprocessed_data, yucca_raw_data
+from yucca.preprocessing.normalization import normalizer
+from yuccalib.utils.nib_utils import get_nib_spacing, get_nib_orientation, reorient_nib_image
+from yuccalib.utils.type_conversions import nifti_or_np_to_np, read_file_to_nifti_or_np
+from yuccalib.image_processing.objects.BoundingBox import get_bbox_for_foreground
+from yuccalib.image_processing.cropping_and_padding import crop_to_box, pad_to_size
+from multiprocessing import Pool
+from skimage.transform import resize
+from batchgenerators.utilities.file_and_folder_operations import (
+    join,
+    load_json,
+    subfiles,
+    save_pickle,
+    maybe_mkdir_p,
+    isfile,
+    subdirs,
+)
+
+
+class UnsupervisedPreprocessor(YuccaPreprocessor):
+    def initialize_paths(self):
+        super().initialize_paths()
+        # Have to overwrite how we get the subject_ids as there's no labelsTr to get them from.
+        # Therefore we use the imagesTr folder and remove the modality suffix.
+        self.subject_ids = subfiles(join(self.input_dir, "imagesTr"), join=False)
+        self.subject_ids = [re.sub(r"_\d+\.", ".", subject) for subject in self.subject_ids]
+
+    def _preprocess_train_subject(self, subject_id):
+        image_props = {}
+        subject_id = subject_id.split(os.extsep, 1)[0]
+        print(f"Preprocessing: {subject_id}")
+        arraypath = join(self.target_dir, subject_id + ".npy")
+        picklepath = join(self.target_dir, subject_id + ".pkl")
+
+        if isfile(arraypath) and isfile(picklepath):
+            print(f"Case: {subject_id} already exists. Skipping.")
+            return
+        # First find relevant images by their paths and save them in the image property pickle
+        # Then load them as images
+        # The '_' in the end is to avoid treating Case_4_000 AND Case_42_000 as different versions
+        # of the label named Case_4 as both would start with "Case_4", however only the correct one is
+        # followed by an underscore
+        imagepaths = [impath for impath in self.imagepaths if os.path.split(impath)[-1].startswith(subject_id + "_")]
+
+        image_props["image files"] = imagepaths
+        images = [read_file_to_nifti_or_np(image) for image in imagepaths]
+
+        if not self.disable_unittests:
```

Comment:
I would probably change the naming of this variable to `self.sanity_check` to avoid confusion with unit tests.